File: app/src/main/java/com/xcyh/reachmax/main/task/TaskCenterActivity.java
package com.xcyh.reachmax.main.task;

import android.content.Context;
import android.os.Bundle;
import android.view.View;

import com.baidu.baselibrary.base.BaseActivity;
import com.baidu.baselibrary.request.EmptyPresenter;
import com.baidu.baselibrary.util.ui.AnimationUtil;
import com.xcyh.reachmax.R;
import com.xcyh.reachmax.databinding.ActivityTaskCenterBinding;
import com.xcyh.reachmax.app.meta.utils.ViewPagerHelper2;
import com.gyf.immersionbar.ImmersionBar;
import com.xcyh.reachmax.adv.ViewUtil;
import com.xcyh.reachmax.model.constant.TaskStatus;

import net.lucode.hackware.magicindicator.buildins.commonnavigator.CommonNavigator;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.CommonNavigatorAdapter;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.IPagerIndicator;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.IPagerTitleView;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;

/**
 * Created by haojiangfeng on 2024/11/12.
 */
public class TaskCenterActivity extends BaseActivity<ActivityTaskCenterBinding, EmptyPresenter>  {

    private final List<String> mTabList = new ArrayList<>();
    private final List<Fragment> mFragments = new ArrayList<>();
    private FragmentStateAdapter mAdapter;



    @Override
    protected int getLayoutId() {
        return R.layout.activity_task_center;
    }

    @Override
    protected void initTitleBar(Bundle savedInstanceState) {
        mRootBinding.titleBar.setTitle("任务中心");
        ImmersionBar.with(this)
                .statusBarColor(R.color.white)
                .autoStatusBarDarkModeEnable(true)
                .navigationBarColor(R.color.white)
                .autoNavigationBarDarkModeEnable(true)
                .init();
    }

    @Override
    protected void initListeners() {

    }

    @Override
    protected void initData() {

    }

    @Override
    protected void initView() {
        super.initView();
        initViewPager();
        initTabLayout();
    }

    private void initViewPager() {
        mFragments.add(com.xcyh.reachmax.main.task.TaskListFragment.getInstance(TaskStatus.ALL));
        mFragments.add(com.xcyh.reachmax.main.task.TaskListFragment.getInstance(TaskStatus.NOT_START));
        mFragments.add(com.xcyh.reachmax.main.task.TaskListFragment.getInstance(TaskStatus.FINISH));
        mFragments.add(com.xcyh.reachmax.main.task.TaskListFragment.getInstance(TaskStatus.CANCEL));
        mAdapter = new FragmentStateAdapter(this) {

            @Override
            public int getItemCount() {
                return mFragments.size();
            }

            @NonNull
            @Override
            public Fragment createFragment(int position) {
                return mFragments.get(position);
            }
        };
        mBinding.viewPager.setAdapter(mAdapter);
    }

    private void initTabLayout() {
        mTabList.add("全部");
        mTabList.add("未开始");
        mTabList.add("已完成");
        mTabList.add("已取消");

        // Tab适配器、设置渐变色、indicator拉伸效果
        CommonNavigator commonNavigator = new CommonNavigator(this);
        commonNavigator.setAdjustMode(false);
        CommonNavigatorAdapter mCommonNavigatorAdapter = new CommonNavigatorAdapter() {
            @Override
            public int getCount() {
                return mTabList.size();
            }

            @Override
            public IPagerTitleView getTitleView(Context context, int index) {
                return ViewUtil.getPagerTitleView(TaskCenterActivity.this, mTabList.get(index), new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        mBinding.viewPager.setCurrentItem(index, false);
                    }
                });
            }

            @Override
            public IPagerIndicator getIndicator(Context context) {
                return ViewUtil.getPagerIndicator(TaskCenterActivity.this);
            }
        };
        commonNavigator.setAdapter(mCommonNavigatorAdapter);
        mBinding.magicIndicator.setNavigator(commonNavigator);

        ViewPagerHelper2.bind(mBinding.magicIndicator, mBinding.viewPager);
    }


    @Override
    public void finish() {
        super.finish();
        AnimationUtil.overridePendingTransition(this, R.anim.anim_none, R.anim.push_right_out);
    }
}

File: app/src/main/java/com/xcyh/reachmax/main/task/TaskCenterActivity.kt
package com.xcyh.reachmax.main.task

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import com.baidu.baselibrary.util.ui.AnimationUtil
import com.gyf.immersionbar.ImmersionBar
import com.tencent.common.util.ToastUtils
import com.xcyh.reachmax.R
import com.xcyh.reachmax.main.task.compose.TaskCenterScreen
import com.xcyh.reachmax.model.bean.task.TaskBean

/**
 * 任务中心（Compose 实现）。
 *
 * 原类继承 BaseActivity<ActivityTaskCenterBinding, EmptyPresenter>，用 MagicIndicator +
 * ViewPager2 + 4 个 TaskListFragment 渲染；现改为 ComponentActivity，用 Compose 渲染、
 * ViewModel 管理状态、[com.base.net.request.BaseRequest] 复用网络层。
 * 类名/包路径保持不变，外部引用（MineFragment、AndroidManifest）无需改动。
 */
class TaskCenterActivity : ComponentActivity() {

    private val viewModel: TaskCenterViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupImmersive()
        setContent {
            val state by viewModel.uiState.collectAsState()
            var pendingCancelTask by remember { mutableStateOf<TaskBean?>(null) }

            LaunchedEffect(Unit) {
                viewModel.events.collect { event ->
                    when (event) {
                        is TaskCenterEvent.Toast -> ToastUtils.showCenterShort(event.message)
                        is TaskCenterEvent.ShowCancelConfirm -> pendingCancelTask = event.task
                    }
                }
            }

            TaskCenterScreen(
                state = state,
                onTabSelected = viewModel::selectTab,
                onRefresh = viewModel::refresh,
                onLoadMore = viewModel::loadMore,
                onCancelClick = viewModel::requestCancel,
                onBack = { finish() },
            )

            pendingCancelTask?.let { task ->
                AlertDialog(
                    onDismissRequest = { pendingCancelTask = null },
                    title = { Text("提示") },
                    text = { Text("确定要取消任务吗？") },
                    confirmButton = {
                        TextButton(onClick = {
                            pendingCancelTask = null
                            viewModel.cancelTask(task)
                        }) { Text("确定") }
                    },
                    dismissButton = {
                        TextButton(onClick = { pendingCancelTask = null }) {
                            Text("取消")
                        }
                    },
                )
            }
        }
    }

    private fun setupImmersive() {
        ImmersionBar.with(this)
            .reset()
            .transparentStatusBar()
            .transparentNavigationBar()
            .statusBarDarkFont(true)
            .navigationBarDarkIcon(true)
            .init()
    }

    override fun finish() {
        super.finish()
        AnimationUtil.overridePendingTransition(this, R.anim.anim_none, R.anim.push_right_out)
    }
}
